models/models.go

```diff
@@ -176,6 +178,22 @@ func (r *Route) loadUser(db *gorm.DB) (utils.User, error) {
 	return LoadUser(userData)
 }
 
+func (r *Route) GetHeaders() (headers utils.Headers) {
```

Comment:
This should be `AddHeaders` since the action is adding them as text to the existing header protos.

Reply:
what this is doing (or what I hope it's doing, at least) is creating a `utils.Headers`, then adding all of the `header`s from `r.headers` to it, then returning it.

Reply:
The logic should be split up into two functions (or refactored to go somewhere else), it's generally bad practice to lump set logic in a get function.

Reply:
I've refactored it out, but I don't see this as mixing set and get logic. this is exposing data on a `Route` in as a different type than it is stored in order to make it easier to conform to APIs that expect `utils.Headers`.

Reply:
Gotcha, looks great!